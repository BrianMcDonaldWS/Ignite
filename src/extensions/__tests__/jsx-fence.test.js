import MarkdownIt from 'markdown-it';
import jsx from 'markdown-it-jsx';
import fence from '../jsx-fence';

// Fence rule must come before jsx rule
// Our fence rule removes the surrounding {' '} for code highlighting
const md = new MarkdownIt().use(fence).use(jsx);

test('jsx - fence', () => {
  expect(md.render('<Foo />')).toMatchSnapshot();
  expect(md.render('```# something in a code block```')).toMatchSnapshot();
});
