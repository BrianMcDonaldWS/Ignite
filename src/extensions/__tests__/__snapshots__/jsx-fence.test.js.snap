// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsx - fence 1`] = `
"<Foo />
"
`;

exports[`jsx - fence 2`] = `
"<p><code># something in a code block</code></p>
"
`;
